tests/gtest/border_agent_callback_test.cpp

```diff
@@ -0,0 +1,273 @@
+/*
+ *  Copyright (c) 2025, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <map>
+
+#include <openthread/border_agent.h>
+#include <openthread/dataset.h>
+#include <openthread/dataset_ftd.h>
+#include <openthread/error.h>
+#include <openthread/instance.h>
+#include <openthread/ip6.h>
+#include <openthread/thread.h>
+
+#include "fake_platform.hpp"
+
+using namespace ot;
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_1
+static const char kThreadVersionString[] = "1.1.1";
+#elif OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_2
+static const char kThreadVersionString[] = "1.2.0";
+#elif OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_3
+static const char kThreadVersionString[] = "1.3.0";
+#elif OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_3_1
+static const char kThreadVersionString[] = "1.3.1";
+#elif OPENTHREAD_CONFIG_THREAD_VERSION == OT_THREAD_VERSION_1_4
+static const char kThreadVersionString[] = "1.4.0";
+#endif
+
+using TxtEntryMap = std::map<std::string, std::vector<uint8_t>>;
+
+TxtEntryMap ParseTxtEntires(const std::vector<uint8_t> &aTxtData)
```

Comment:
Thanks for adding the test for this.

I suggest adding it to the existing `nexus/test_border_agent.cpp` test. 
- This way, all border agent tests and behaviors are validated within the same test. 
- We can use all the OpenThread core methods and helpers. For example, can use the existing TXT entry parser, `TxtEntry::Iterator::GetNextEntry()`, and don't need to implement it within the test again. 
- Also having access to the OpenThread core and C++ types can help make code for changing the dataset (changing the network name, etc.) simpler.

Generally, the gtest (fake platform) tests are better suited for RCP/radio-related tests. For tests targeting specific modules, unit tests (either direct or using the Nexus) make more sense.

Reply:
Thanks! These points all make sense to me. I've updated to write the test using nexus.

BTW, there are two things I like about gtest (We may consider to bring them into nexus framework later):
* We can use stl. Generally I feel use stl (like vector, map) is very handy to write test cases.
    * When using nexus, since we overload the `new` operator, we cannot use stl. Not sure if this issue can be resolved.
* `--gtest_filter` is useful for me to run only part of the test cases in the test binary. Though in most time it just spends a few seconds to run all the test cases, it would be great to solve this small pain point :)